backend/de.metas.manufacturing/src/main/java/de/metas/manufacturing/generatedcomponents/MacAddressGenerator.java

```diff
@@ -79,9 +73,11 @@ public MacAddressGenerator()
 	@Override
 	public ImmutableAttributeSet generate(@NonNull final ComponentGeneratorContext context)
 	{
+		final ImmutableSet<AttributeCode> supportedAttributes = attributeDAO.getAttributeCodesThatStartWith(ATTR_CODE_PREFIX_RouterMAC);
```

Comment:
IMHO you have to order them by their numeric index, else the attributes will be allocated in a pseudo-random order.

something like, for each attribute code, take the substring after ATTR_CODE_PREFIX_RouterMAC, right pad it to 10 chars, fill with 0 and them order by that string.
e.g.
* for RouterMAC1  -> subtring is 1, right padded would be 000000001
* * for RouterMAC10  -> subtring is 10, right padded would be 000000010

Reply:
thx, pushed